service/controller/cluster.go

```diff
@@ -94,13 +121,56 @@ func NewCluster(config ClusterConfig) (*Cluster, error) {
 		c := controller.Config{
 			K8sClient: config.K8sClient,
 			Logger:    config.Logger,
-			Name:      config.ProjectName,
-			ResourceSets: []*controller.ResourceSet{
-				resourceSet,
-			},
+			Name:      project.Name(),
+			Resources: resources,
 			NewRuntimeObjectFunc: func() runtime.Object {
 				return new(v1alpha1.AzureConfig)
 			},
+			InitCtx: func(ctx context.Context, obj interface{}) (context.Context, error) {
```

Comment:
Do you have plans to get rid of the controller context usage? Further, this function should not do more than set the controller context so that it is initialized. All other logic should happen somewhere else. [In the `aws-operator` this is not more than 1 line](https://github.com/giantswarm/aws-operator/blob/762d8a8ce09be1d7ea95815f08a0e6c1fb63e070/service/controller/cluster.go#L112-L114).

Reply:
Yes, we will tackle that in the near future.